lib/ansible/module_utils/docker_common.py

```diff
@@ -589,6 +590,20 @@ def pull_image(self, name, tag="latest"):
 
         return new_tag, old_tag == new_tag
 
+    def report_warnings(self, result, warnings_key=None):
+        '''
+        Checks result of client operation for warnings, and if present, outputs them.
+        '''
```

Comment:
Probably a good idea to add some examples of what data you expect result to contain here and also document warnings_key.  It will make the algorithm more readily understandable.

Reply:
I'm still in the process of finding that out. The docker-py API is not very well documented, in fact the warnings I added are the only two which are documented. According to the daemon API documentation, some other endpoints also return warnings (though usually at most one warning), but in some cases these are not passed on by docker-py. Once I'll figure out some more I'll extend the original code...